controller/linodemachine_controller_helpers.go

```diff
@@ -107,20 +115,59 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 	return createConfig, nil
 }
 
-func buildInstanceAddrs(linodeInstance *linodego.Instance) []clusterv1.MachineAddress {
+func (r *LinodeMachineReconciler) buildInstanceAddrs(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, instanceID int) ([]clusterv1.MachineAddress, error) {
 	addrs := []clusterv1.MachineAddress{}
-	for _, addr := range linodeInstance.IPv4 {
-		addrType := clusterv1.MachineExternalIP
-		if addr.IsPrivate() {
-			addrType = clusterv1.MachineInternalIP
-		}
+	ips, err := r.getInstanceIPv4Addresses(ctx, logger, machineScope, instanceID)
+	if err != nil {
+		logger.Error(err, "Failed to get instance ip addresses")
+		return nil, err
+	}
+
+	// add all instance ips to machine's status
+	for _, ip := range ips {
 		addrs = append(addrs, clusterv1.MachineAddress{
-			Type:    addrType,
-			Address: addr.String(),
+			Type:    ip.ipType,
+			Address: ip.ip,
 		})
 	}
 
-	return addrs
+	return addrs, nil
+}
+
+func (r *LinodeMachineReconciler) getInstanceIPv4Addresses(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, instanceID int) ([]nodeIP, error) {
+	addresses, err := machineScope.LinodeClient.GetInstanceIPAddresses(ctx, instanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// get the default instance config
+	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, instanceID, &linodego.ListOptions{})
+	if err != nil || len(configs) == 0 {
+		logger.Error(err, "Failed to list instance configs")
+		return nil, err
```

Comment:
nit: 
```suggestion
		return nil, fmt.Errorf("list instance configs: %w", err)
```
For helper functions like this and in the case where you're both logging and returning an error, you can surface the context of an error with `fmt.Errorf`